compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/ContributedInterfaceSupertypeGenerator.kt

```diff
@@ -118,6 +121,30 @@ internal class ContributedInterfaceSupertypeGenerator(session: FirSession) :
       }
     }
 
+  private fun FirRegularClassSymbol.contributesAnnotations(): Sequence<FirAnnotation> {
+    return annotations.annotationsIn(session, session.classIds.allContributesAnnotations).ifEmpty {
+      if (origin == FirDeclarationOrigin.Library) {
```

Comment:
@bnorm question for later - is this the right way in FIR to check if it's from an external source?